WordPress/src/main/java/org/wordpress/android/ui/comments/CommentsDetailActivity.java

```diff
@@ -0,0 +1,250 @@
+        if (item.getItemId() == android.R.id.home) {
+            finish();
+            return true;
+        }
+        return super.onOptionsItemSelected(item);
+    }
+
+    @Override
+    public void onLoadMore() {
+        updateComments();
+    }
+
+    private void updateComments() {
+        if (mIsUpdatingComments) {
+            AppLog.w(AppLog.T.COMMENTS, "update comments task already running");
+            return;
+        } else if (!NetworkUtils.isNetworkAvailable(this)) {
+            ToastUtils.showToast(this, getString(R.string.error_refresh_comments_showing_older));
+            return;
+        } else if (!mCanLoadMoreComments) {
+            AppLog.w(AppLog.T.COMMENTS, "no more comments to be loaded");
+            return;
+        }
+
+        final int offset = mAdapter.getCount();
+        mDispatcher.dispatch(CommentActionBuilder.newFetchCommentsAction(
+                new FetchCommentsPayload(mSite, mStatusFilter, COMMENTS_PER_PAGE, offset)));
+        mIsUpdatingComments = true;
+        setLoadingState(true);
+    }
+
+    @SuppressWarnings("unused")
+    @Subscribe(threadMode = ThreadMode.MAIN)
+    public void onCommentChanged(CommentStore.OnCommentChanged event) {
+        mIsUpdatingComments = false;
+        setLoadingState(false);
+        // Don't refresh the list on push, we already updated comments
+        if (event.causeOfChange != CommentAction.PUSH_COMMENT) {
+            if (event.changedCommentsLocalIds.size() > 0) {
+                loadDataInViewPager();
+            } else if (!event.isError()) {
+                //There are no more comments to load
+                mCanLoadMoreComments = false;
+            }
+        }
+        if (event.isError()) {
+            if (!TextUtils.isEmpty(event.error.message)) {
+                ToastUtils.showToast(this, event.error.message);
+            }
+        }
+    }
+
+    private void loadDataInViewPager() {
+        if (mIsLoadingComments) {
+            AppLog.w(AppLog.T.COMMENTS, "load comments task already active");
+        } else {
+            new LoadCommentsTask(mCommentStore, mStatusFilter, mSite, new LoadCommentsTask.LoadingCallback() {
+                @Override
+                public void isLoading(boolean loading) {
+                    setLoadingState(loading);
+                    mIsLoadingComments = loading;
+                }
+
+                @Override
+                public void loadingFinished(CommentList commentList) {
+                    if (!commentList.isEmpty()) {
+                        showCommentList(commentList);
+                    }
+                }
+            }).executeOnExecutor(AsyncTask.THREAD_POOL_EXECUTOR);
+        }
+    }
+
+    private void showCommentList(CommentList commentList) {
+        final int previousItem = mViewPager.getCurrentItem();
+
+        //Only notify adapter when loading new page
+        if (mAdapter != null && mAdapter.isAddingNewComments(commentList)) {
```

Comment:
This solution only adds items and notifies the adapter when we're loading the next page. Otherwise it rebuilds the adapter (for example on editting a comment). It is not perfect but we can't easily tell the FragmentStatePagerAdapter to reload a fragment (all the solutions I found are sort of a hacks). On the other hand the only con for a user is that the page blinks on reload.